### ISSUE TITLE
Disable follower accounts or bucket them

### ISSUE THREAD

**Issue by nolanlawson** [tootcafe/discussions#16]
Based on discussion in [this thread](https://toot.cafe/users/nolan/updates/4839), I would like to either disable follower accounts or (more likely) shift to a bucket-based system.

Read the full thread for context, but I am tired of the hero worship, celebrity culture, and (yes) #branding that emerges from public follower counts. On the other hand, they're a good metric for distinguishing trolls (and newbies you want to welcome!) from established users.

My recommendation:

1. You can see your own follower count
2. Publicly, follower counters are displayed normally until you get to >100, at which point it just says "100+"

Thoughts? 🙂

**Comment by mlcdf** [comment]
I stumble on this thread a couple hours ago, and I had the same idea. So yeah, huge :+1:

**Comment by nraboy** [comment]
I personally would like to see a follower count remain.  Helps me determine a few things:

* Is this person legit or a knock-off account.  With so many instances and people claiming to be Mark Zuckerberg, a clear indicator would be determinable by their follower count.
* Do others find this person to share valuable information.  Few followers is generally a good indicator the person doesn't share anything worth following.

If you absolutely have to disable public followers, maybe you can consider doing like 1000+ rather than 100+?

**Comment by halkeye** [comment]
We have follower counts? I guess I always assumed we did.

On Apr 12, 2017 10:38 AM, "Maxime Le Conte des Floris" <
notifications@github.com> wrote:

I stumble on this thread earlier, and I had the same idea. So yeah, huge 👍

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/tootcafe/discussions/issues/16#issuecomment-293653236>,
or mute the thread

**Comment by dperrera** [comment]
I love this idea. Removing follower counts disincentivizes bad behavior and would quash some of our reflexive admiration of big numbers. Hopefully this move would focus people's attention on good content and thoughtful conversation. At the very least it would be an excellent experiment.

**Comment by newtoncalegari** [comment]
+1 for your suggestion. 
As said in the thread, follower counts can eventually create a kind of unconscious bias, where we might judge people according to those numbers instead of paying attention to their actual content.
Agree with @dperrera, it would be a nice experiment

**Comment by colinhowells** [comment]
To nraboy's point: follower numbers would be the only user verification mechanism we had – if the instance's admin weren't acting as gatekeeper. Since Nolan's acting as such for the cafe, follower numbers could be dropped, for us. Other larger, more open instances might still need the number as people in them would need all the help they could get ...

**Comment by moke-codes** [comment]
I agree with the "unconscious bias" thing and the role that the followers' count plays in that matter.
My only question is, how can someone know that the person he found during a search is the person it's looking for and not someone else pretending to be that person?

**Comment by mlcdf** [comment]
@nraboy

> Is this person legit or a knock-off account. With so many instances and people claiming to be Mark Zuckerberg, a clear indicator would be determinable by their follower count.

There are other ways to authenticate an account. So for me, this is a none issue.

> Do others find this person to share valuable information. Few followers is generally a good indicator the person doesn't share anything worth following.

Well, "anything worth following" is pretty subjective. 

In order to know if a person share anything worth following, the best way is by far to read what the person posts. To me, the follower count is just a "fame" indicator, not a "quality" indicator.

**Comment by blakeembrey** [comment]
@Moke-RS Isn't that an issue regardless of follower counts being visible? I could be wrong, but I thought you'd only see federated followers (if you're using search), in which case a legit user could easily have less followers than than a fake user from your perspective.

Either way, that's only accounting for other instances. On toot.cafe, as I understand it, it's mostly a curated community so it really wouldn't matter. The people on here should be real, and following the other discussions, fake users should be removed.

Edit: Other discussion mentioned is https://github.com/tootcafe/discussions/issues/9.

**Comment by nraboy** [comment]
@blakeembrey but my interests are not in strictly JavaScript.  If I want to adventure out and follow someone who focuses on hardware, how can I be sure that they are real or not?  Likewise if someone from another community wants to follow someone from a JavaScript community.  Do we have to trust that JavaScript people stay in the JavaScript community and nowhere else? That seems not right.

**Comment by moke-codes** [comment]
> @Moke-RS Isn't that an issue regardless of follower counts being visible?

@blakeembrey Yes, I think so. I'm not advocating for the follower's counts, just raising a concern. How could we tackle this problem? I don't know the answer for this. Also, I'm not sure if a "certified account" won't be necessary at any moment in the future.

**Comment by blakeembrey** [comment]
@nraboy Please check the first paragraph of my response. I'm not advocating either way, I'm just saying that using the count as a way to "prove" the user is real seems flawed. If you're viewing people on other communities, then you'd see their count on that community, and you can use that. If you're using the federated follower count from toot.cafe, seems pretty much irrelevant. I'd recommend using more real signals, and in both of these cases disabling it on toot.cafe seems like a win rather than being bad (what one community does wouldn't affect anyone else, nor does it affect you to find those people you'd like to follow in other communities).

**Comment by halkeye** [comment]
Just for the record this would only affect the profiles of toot.cafe users. I'm pretty sure the profile of anyone else goes to the server they are on. If we do have a local copy, the followers and stuff is wrong anyways.

**Comment by nraboy** [comment]
@blakeembrey Maybe I don't understand the whole thing.  Are you saying local follower count and federated follower count are two different things? As in, if federated was disabled, anyone who clicks my name would be able to see how many local toot.cafe people follow me?

I think I see where this is going, but I still have skepticism.

In all honesty, I came to Mastodon to dodge all the political and hate related content that showed up on the other social media sites.  I prefer to have follower counts and don't really care if it creates a "god mode" atmosphere.  I know everyone came for different reasons and I know that you can't satisfy everyone, but I think most came because of the same reason I came.  Wouldn't it be a bit much to worry about followers as of right now?

**Comment by blakeembrey** [comment]
Here's a real example that hopefully makes more sense, if you'd like to verify it and see that the indicator really doesn't mean much when you're looking at people outside the current instance. Take a look at https://mastodon.social/@danharmon, then go to toot.cafe and paste it into search instead:

![image](https://cloud.githubusercontent.com/assets/1088987/24972968/062a1266-1f72-11e7-82fa-5a3ccf6a1845.png)

Hmm, that user doesn't look very real now. No posts and only four followers? But if you opened their profile directly, more realistic:

![image](https://cloud.githubusercontent.com/assets/1088987/24973061/47fc3854-1f72-11e7-85bf-af2b70730695.png)

As a closed community, I don't see any value from displaying follower counts.

The real answer to this, as @Moke-RS mentioned, is likely something else. These indicators need to come from somewhere else in a federated social network. My current thought is the only real indicator can come from your other public identities (e.g. blogs, existing social media accounts, etc).

**Comment by blakeembrey** [comment]
@nraboy Of course, I'm only touching on the comments around using follower count and search as an indicator of "real" to demonstrate that, in that case, it's irrelevant and not a good metric. For other reasons it might be useful (you mentioned whether the content had quality) but maybe that's not needed. Maybe people prefer to read through their previous content to see if it interests them? Personally, I don't think follower counts, given the federated drawbacks, are a good measure of either of these features, but I would love to see myself proven wrong. 

Edit: For finding people and content to follow that's valuable, I feel like it's a little bit of an unknown problem in Mastodon so far. I'm sure an organic solution will evolve, but for now I'd use indicators that you can see - their past content, existing social networks/blogs and their followers if their on another instance that's less restrictive (e.g. anyone can join). If toot.cafe does disable follower counts, it might be worth adding an indicator somehow that it's a "closed community and not just anyone can join" so that people externally visiting can see why there's no count quickly.

**Comment by nraboy** [comment]
I understand.

Maybe just keep it as close to the upstream as possible?  Model it towards mastodon.social and maybe influence how the upstream does things so it trickles down to the other instances?  This way it doesn't become chaos between all the instance differences?

**Comment by ef4** [comment]
I agree with the sentiment that follower counts are a rough proxy for identity authenticity. If we were to drop them, I think some other social-graph-derived indicator is needed. Perhaps "$PERSON is followed by these people you follow".

I don't think being a closed instance makes a difference -- being federated is important, and users should get reasonable feedback about whoever they're viewing whether or not their home instance is elsewhere. (This is already a weak spot in the UX -- it would be better to not show any counts than to show the local-only counts that are present now, which are actively misleading)

**Comment by kkostov** [comment]
> You can see your own follower count

Sounds like a good idea.

> Publicly, follower counters are displayed normally until you get to >100, at which point it just says "100+"

I am not sure this would work very well. LinkedIn has a similar system where you normally see up to 500 and then it simply says a person has 500+ connections. I know the context there is different but ultimately the meaning of "100+" would be diminished as it would only apply to new accounts or users who simply prefer to browse without actually posting enough content to gain followers. Everyone else will be "100+".

Since the community is invite-only, we could assert that people are good by default and report the occasional bot/troll instead. 

I am also uncertain why do we need to focus so much on confirming someone's real identity. It's the internet... if they want to be Mark Z., let them be. If someone is misbehaving they will be blocked / reported and eventually removed.

My proposal would be to place a badge like "new comer"* on the account.
It would go away once the user has a few toots which have been boosted at least once by someone who is not a "new comer". Nothing too severe - we could run some statistics on the current user activity to determine a good value for the number of boosted toots needed to remove the badge.

* the name should be catchy and not suggest you are a "newbie" or anything like that.

**Comment by Pajn** [comment]
I think it's a pretty neat idea, however I think it should be done completely if it is done.
The reason is that < 100/> 100 has a pretty big risk of encouraging the "validated" feeling based on followers, as we give an exact point where the account has reached "many" followers. While this feeling might feel nice it's problematic as it doesn't mean anything really. If a fake account is created first and start posting nice stuff it will get followers and if the real person later registers it will have fewer followers.

It's better to take away that problem entirely and leave validation keybase and other actual solutions for that.

I also don't think it's needed to discover new users as the post count tells you that as well.

**Comment by moke-codes** [comment]
> I agree with the sentiment that follower counts are a rough proxy for identity authenticity. If we were to drop them, I think some other social-graph-derived indicator is needed. Perhaps "$PERSON is followed by these people you follow".

I like this approach. You can hide the followers' count, but show the "most followed people who follow this person + people you know who follow this person". Prominent people used to be followed by other prominent people. It's an indirect way of giving credibility.

**Comment by nraboy** [comment]
@Moke-RS I also like this as well :-).  Anyone want to write this, it seems complicated :-P

**Comment by moke-codes** [comment]
@nraboy (List of followers ordered by its followers' count [which is hidden but recorded] + List of common followers) - I don't how much of this is complicated to do. It depends, I guess, on how this information flow between instances.

**Comment by blakeembrey** [comment]
Should we make an issue on Mastodon for this? It seems like it should be doable, but from the little I've used Mastodon it seems like it'd need to request more information than available today. Twitter and Facebook both use the same feature, and I think I've referred to that more often than the actual number of followers/friends.

Edit: If I understand it correctly, it should be enough just to find a way to pull the remote "following" list. Then intersect that with the people I'm following to get the list of mutual followers.

**Comment by nolanlawson** [comment]
For those wondering, the UI doesn't show the full remote user metrics (follower count etc.) for the same reason it doesn't necessarily show all boosts/replies/favs: it's only when someone in our instance _starts following_ someone in another instance that we begin to receive information about them. So for instance Dan Harmon has "4 followers" because he has 4 followers that are followed by other users in our instance.

I'm not sure if there's an open issue to change this, but it's technically feasible (assuming no CORS issues) although I'm not sure it's within the project goals. Worth looking around the https://github.com/tootsuite/mastodon GitHub repo.

As for distinguishing fake vs genuine accounts, we have an open issue on this (#9) but based on discussion it seems we will just ban fake accounts on our instance which makes the point moot.

Also it's good to recall that we can only do this for our _own_ instance; follower counts will still be visible on other instances unless, say, someone submits a patch to Mastodon to allow for follower counts to be capped at _n_ followers and then other instances decide to follow our model.

@kkostov makes a good point, that effectively every active user would be 100+, but I guess in that case I would prefer to make the number 10 instead of 100 because at least then you can tell when someone _just_ made an account and thus might need some help getting shown around. :)

It sounds like step 1 is a patch to Mastodon to allow admins to disable follower count numbers (at the number of their choosing), and then we can decide whether we want it to be 100 or 50 or 10 or 0. Does that make sense?

**Comment by nolanlawson** [comment]
[More discussion here](https://mastodon.social/@GinnyMcQueen/2373876).

**Comment by schamp** [comment]
I like the idea of finding some other quality indicator besides raw follower count, but I'm thinking, for instances that remain invitation-only, it's probably not so important.  "High follower count followers" seems like a good starting point.

**Comment by nolanlawson** [comment]
So I saw some nasty subtooting going on recently, and it's made me feel even more strongly that follower counts needs to go away. Ignoring the specifics of the situation, I believe the ultimate cause is someone thinking to themselves, "This person has _influence_! And yet they're _wrong_! I must correct this injustice!" Whereas if you have no idea how much "influence" someone says, you don't feel like youn need to go a righteous mission to harass them and tell all your followers to gang up on them.
